clients/upssched.c

```diff
@@ -757,30 +747,39 @@ static void sendcmd(const char *cmd, const char *arg1, const char *arg2)
 
 		/* we're connected now */
 
-		ret = write(pipefd, enc, enclen);
+		ret = write(pipefd, enc, sizeof(enc));
```

Comment:
Got to a computer now for another look at the code: the contents of `buf` and `enc` are prepared by a series of `snprintf`'s, so they contain some string that is no longer than the buffer size - but may be smaller. So writing the "string, null-char, garbage" padding to the buffer size in this line's change may be pointless at best, unsafe somehow at worst.

So I'm in favor of fixing this bit back to `enclen`/`buflen` equaling the respective `strlen()` result, and will post a commit shortly.